**Bioinformatics, genomics, agriculture, food science, medicine, genetic engineering, etc...**

* [BIOINFORMATICS](#bioinformatics)
* [GENOMICS](#genomics)
* [NEUROSCIENCE](#neuroscience)
   * [Neural-Networks](#neural-networks)
* [STRUCTURAL-BIOLOGY](#structural-biology)

   
# BIOINFORMATICS
* [epitopes](https://github.com/hammerlab/epitopes) → Python interface to immunology and bioinformatics datasets (i.e. IEDB, cancer antigens, TCGA mutant peptides).
* biostar-central → [The software that runs the Biostars Bioinformatics Q&A at: http://www.biostars.org](https://github.com/ialbert/biostar-central)
* screed (GED@MSU.EDU) → [A simple read-only sequence database, designed for short reads](https://github.com/ged-lab/screed) from the Genomics, Evolution, and Development Lab. [Documentation](http://readthedocs.org/docs/screed/en/latest/)



# GENOMICS
* BreakSeq2 : [Ultrafast and accurate nucleotide-resolution analysis of structural variants](https://github.com/bioinform/breakseq2)
* CompleteGenomicsTools → [Complete Genomics provides whole-genome sequencing using DNA nanoball arrayed sequencing ](https://github.com/seandavi/CompleteGenomicsTools). Software for manipulating and visualizing Complete Genomics data, with a focus on cancer: http://watson.nci.nih.gov/~sdavis/software/cgent/
* [Distributed Evolutionary Algorithms in Python (EA)](http://code.google.com/p/deap/)
* Genetic Programming → [Evolution of Mona Lisa](http://rogeralsing.com/2008/12/07/genetic-programming-evolution-of-mona-lisa/). Here is the [non-Python LGPL licensed source code](http://code.google.com/p/alsing/downloads/list), which should make an interesting project to port to Python. 
   * [Pierre Lindenbaum's implementation of the Genetic Algorithm described by Roger Alsing](http://plindenbaum.blogspot.in/2008/12/random-notes-2008-12.html) above.
* [Harvest](https://github.com/marbl/harvest) is a suite of core-genome alignment and visualization tools for quickly analyzing thousands of intraspecific microbial genomes. It includes Parsnp, a fast core-genome multi-aligner, and Gingr, a dynamic visual platform. Combined they provide interactive core-genome alignments, variant calls, recombination detection, and phylogenetic trees.
* MetaSVMerge :: [An accurate method-aware merging algorithm for structural variations](https://github.com/bioinform/metasv)
* [pygenetorrent](https://github.com/hammer/pygenetorrent) → A Python client for GeneTorrent.


##### Resources
* Notes: [An introductory approach (with an application to the Prisoner's Dilemma) to Genetic Algorithms](http://www2.econ.iastate.edu/tesfatsi/holland.gaintro.htm)


# NEUROSCIENCE
* [NIPY](https://github.com/nipy) developers group on GH.
* [BrainImagingPipelines](https://github.com/INCF/BrainImagingPipelines) → Optimized Nipype pipelines for brain imaging.
* [dipy](https://github.com/nipy/dipy) → Diffusion MR Imaging in Python. [Website](http://dipy.org)
* [nipype](https://github.com/nipy/nipype) → Workflows and interfaces for neuroimaging packages <=> [The Nightly build](http://www.mit.edu/~satra/nipype-nightly/)
* [nilearn](http://nilearn.github.io) is a machine learning tool for NeuroImaging in Python. [Source code](https://github.com/nilearn/nilearn)
* [PyBrain](http://pybrain.org) and its [installation](https://github.com/pybrain/pybrain/wiki/installation)
* PyMVPA → [MultiVariate Pattern Analysis in Python](http://www.pymvpa.org). Source repo :: https://github.com/PyMVPA/PyMVPA
* [psychopy](https://github.com/psychopy/psychopy) → For running psychology and neuroscience experiments: http://www.psychopy.org


### Neural Networks

###### Resources
* _Open book_ on [Neural Networks and Deep Learning](http://neuralnetworksanddeeplearning.com/)
   * Chapter-1. [Using neural nets to recognize handwritten digits](http://neuralnetworksanddeeplearning.com/chap1.html)
   * Chapter-2. [How the backpropagation algorithm works](http://neuralnetworksanddeeplearning.com/chap2.html)
   * Chapter-3. Improving the way neural networks learn
   * Chapter-4. Neural nets: the big picture
   * Chapter-5. Deep learning
   * Chapter-6. The state of the art in image recognition
   * Chapter-7. The future of neural nets
* [Deep Neural Networks: A Getting Started Tutorial](http://www.visualstudiomagazine.com/articles/2014/06/01/deep-neural-networks.aspx)   


# STRUCTURAL BIOLOGY
* [CheShift](https://github.com/aloctavodia/cheshift) is a software for prediction of 13Cα and 13Cβ chemical shifts and validation of protein structures.
    

